client/components/LearnerViewDirectiveSpec.js

```diff
@@ -0,0 +1,193 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @LearnerViewDirective Unit tests for the LearnerViewDirective.
+ *
+ */
+
+describe('LearnerViewDirective', function() {
+  var $scope;
+  var element;
+  var QuestionDataService;
+
+  beforeEach(module("tie"));
+
+  var generateRandomChars = function(number) {
+    var generatedChars = "";
+    var possible = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+    for (var i = 0; i < number; i++) {
+      generatedChars += possible.charAt(
+        Math.floor(Math.random() * possible.length));
+    }
+    return generatedChars;
+  };
+
+  var NUM_CHARS_CODE = 100;
+  var LANGUAGE = 'python';
+
+  var SECONDS_TO_MILLISECONDS;
+  var DEFAULT_AUTOSAVE_SECONDS;
+
+  var DELTA_MILLISECONDS = 100;
+
+  var AUTOSAVE_MILLISECONDS;
+
+  var AUTOSAVE_REPEAT_RANGE = 20;
+
+  beforeEach(inject(function($compile, $rootScope, _QuestionDataService_,
+    _SECONDS_TO_MILLISECONDS_, _DEFAULT_AUTOSAVE_SECONDS_) {
+    $scope = $rootScope.$new();
+
+    // The reason why we have to go through this trouble to get $scope
+    // is the controller is anonymous, thus, there is no easy way to do
+    // it. Recommend future refactor.
+    // TODO
```

Comment:
That's not how a TODO works -- it needs a username and should go before the description of work that needs done.. Check other TODOs in the codebase and make sure you're following existing patterns.